sdk/identity/identity/samples/AzureCacheForRedis/node-redis.md

```diff
@@ -0,0 +1,153 @@
+    - [Dependency Requirements](#dependency-requirements-node-redis)
+    - [Authenticate with AAD - Hello World](#authenticate-with-aad-node-redis-hello-world)
+    - [Authenticate with AAD - Handle Re-Authentication](#authenticate-with-aad-handle-re-authentication)
+
+### Node-redis Library
+
+#### Dependency Requirements Node-redis
+```
+    "@azure/identity":"^2.0.4",
+    "redis": "^4.1.0"
+```
+
+
+#### Authenticate with AAD node-redis Hello World
+This sample is intended to assist in authenticating with AAD via node-redis client library. It focuses on displaying the logic required to fetch an AAD Access token and to use it as password when setting up the node-redis instance.
+
+Familiarity with the Node-redis and Azure Identity client libraries is assumed. If you're new to the Azure Identity library for Javascript, see the docs for [Azure Identity](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) and [node-redis](https://github.com/redis/node-redis) rather than this guide.
+
+##### Migration Guidance
+When migrating your existing your application code, you need to replace the password input with Azure Active Directory Token.
+Integrate the logic in your application code to fetch an AAD Access Token via Identity SDK as shown below and replace it with the password configuring/retrieving logic in your application code.
+
+**Note:** The below sample uses `ClientCertificateCredential` from our [Azure Identity](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) SDK, the credential can be replaced with any of the other `TokenCredential` implementations offered by our [Azure Identity](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) SDK.
+
+
+```ts
+import { createClient } from 'redis';
+import * as dotenv from "dotenv";
+import {ClientCertificateCredential} from "@azure/identity";
+dotenv.config();
+
+async function main(){
+// Construct a Token Credential from Identity SDK, e.g. ClientSecretCredential / Client CertificateCredential / ManagedIdentityCredential etc.
+    const credential = new ClientCertificateCredential(process.env.AZURE_TENANT_ID, process.env.AZURE_CLIENT_ID,process.env.CLIENT_CERTIFICATE_PATH)
+    try{
+        // Fetch an AAD token to be used for authentication. This token will be used as the password.
+        let accessToken = await credential.getToken("https://*.cacheinfra.windows.net:10225/appid/.default")
+        console.log("access Token",accessToken);
+
+        //Option 1 - Create redis client and connect to the Azure Cache for Redis over the TLS port using the access token as password.
+        const client = createClient({
+            username: process.env.REDIS_SERVICE_PRINCIPAL_NAME, 
+            password: accessToken.token, 
+            url: `redis://${process.env.REDIS_SERVICE_PRINCIPAL_NAME}:${accessToken.token}@${process.env.REDIS_HOSTNAME}:6380`
```

Comment:
The complete url is needed. The username and password aren't needed if we are providing the url correctly. I did that for clarity purposes so that the users understand node-redis and io-redis. I can remove those fields though

Reply:
Url would work even without the username and password specified in the url. Will update the url.

Reply:
done